lib/ansible/modules/cloud/cloudstack/cs_network_offering.py

```diff
@@ -0,0 +1,379 @@
+  type: string
+  sample: Enabled
+guest_ip_type:
+  description: Guest type of the network offering
+  returned: success
+  type: string
+  sample: Isolated
+availability:
+  description: The availability of network offering
+  returned: success
+  type: string
+  sample: Optional
+service_offering_id:
+  description: The service offering ID
+  returned: success
+  type: string
+  sample: c5f7a5fc-43f8-11e5-a151-feff819cdc9f
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackNetworkOffering(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackNetworkOffering, self).__init__(module)
+        self.returns = {
+            'guestiptype': 'guest_ip_type',
+            'availability': 'availability',
+            'serviceofferingid': 'service_offering_id',
+        }
+        self.network_offering = None
+
+    def get_network_offering(self):
+        if self.network_offering:
+            return self.network_offering
+
+        args = {
+            'name': self.module.params.get('name'),
+            'guestiptype': self.module.params.get('guest_type'),
+        }
+        no = self.cs.listNetworkOfferings(**args)
+        if no:
+            self.network_offering = no['networkoffering'][0]
+
+        return self.network_offering
+
+    def create_or_update(self):
+        network_offering = self.get_network_offering()
+
+        if not network_offering:
+            network_offering = self.create_network_offering()
+
+        return self.update_network_offering(network_offering=network_offering)
+
+    def create_network_offering(self):
+        network_offering = None
+        self.result['changed'] = True
+
+        args = {
+            'state': self.module.params.get('state'),
+            'displaytext': self.module.params.get('display_text'),
+            'guestiptype': self.module.params.get('guest_ip_type'),
+            'name': self.module.params.get('name'),
+            'supportedservices': self.module.params.get('supported_services'),
+            'traffictype': self.module.params.get('traffic_type'),
+            'availability': self.module.params.get('availability'),
+            'conservemode': self.module.params.get('conserve_mode'),
+            'details': self.module.params.get('details'),
+            'egressdefaultpolicy': self.module.params.get('egress_default_policy'),
+            'ispersistent': self.module.params.get('persistent'),
+            'keepaliveenabled': self.module.params.get('keepalive_enabled'),
+            'maxconnections': self.module.params.get('max_connections'),
+            'networkrate': self.module.params.get('network_rate'),
+            'servicecapabilitylist': self.module.params.get('service_capability_list'),
+            'serviceofferingid': self.module.params.get('service_offering_id'),
```

Comment:
for consistency, please use a helper function to query the id from the API by the service offering name if possible.